src/common/types/vector.cpp

```diff
@@ -124,6 +124,41 @@ void Vector::ResetFromCache(const VectorCache &cache) {
 	cache.ResetFromCache(*this);
 }
 
+void Vector::SliceInPlace(Vector &other, idx_t offset, idx_t end) {
```

Comment:
Could we perhaps remove the `SliceInPlace` and additions to `ValidityMask::SliceInPlace ` from this PR? I think it will not be necessary once we add vectorized execution support to `WindowSegmentTree:: Compute` and the code is rather complicated/prone to bugs.